### DISCUSSION TITLE
Size of repository

### DISCUSSION THREAD

**Discussion by detrin** [shap/shap#3207] [General]
I think it is little bit concerning that the size of repository is at this stage
```
~/Documents/projects/temp ❯ git clone https://github.com/shap/shap
Cloning into 'shap'...
remote: Enumerating objects: 14881, done.
remote: Counting objects: 100% (14881/14881), done.
remote: Compressing objects: 100% (4516/4516), done.
remote: Total 14881 (delta 10366), reused 14594 (delta 10238), pack-reused 0
Receiving objects: 100% (14881/14881), 250.30 MiB | 2.45 MiB/s, done.
Resolving deltas: 100% (10366/10366), done.
Updating files: 100% (488/488), done.
```

When I joined my current job, the one repository here had 750 MB, I separated repository into pure code and the rest. I wonder whether it is not a bit too much in this case as well.

**Comment by detrin** [comment]
So for example, the presentations have nothing to do with the repo itself
<img width="504" alt="image" src="https://github.com/shap/shap/assets/18104456/5f8dbf18-fd6a-4703-afa1-61d9b1db26bf">
then .git folder is little bit in my opinion, the same with notebooks
<img width="448" alt="image" src="https://github.com/shap/shap/assets/18104456/6cc29a86-32e0-46a5-995c-fea8f16f989d">
I would question whether we need notebooks within the same repository. Similarly with docs and data.

**Comment by dsgibbons** [comment]
I tend to agree with you. I like that everything is hosted in one place, but it does make cloning more cumbersome than it needs to be.

**Reply by detrin** [reply]
So, what can we do about it? I am not sure what have changed since v0.41.0 regarding ownership and etc. Since I see that now the shape is owned by organization shap. 

My suggestion no. 1:
1. rename standing repository to the old one (e.g. shap_old) 
2. create repository only for source code and tests with only necessary data files (e.g. shap)
3. create new repository for documentation (e.g. shap_docs). Built of documentation would be in the new git repository, but it would clone the code repository (e.g. shap)
4. create new repository for notebooks, user tests, various benchmarks and anything that is nor source code nor docs (e.g. shap_notebooks)
5. after making sure that the whole scheme works deprecate old repository (e.g. shap_old)

My suggestion no. 2:
The same as no one, but not creating new repository instead of that removing master and creating new branch as master. https://medium.com/@sangeethkumar.tvm.kpm/cleaning-up-a-git-repo-for-reducing-the-repository-size-d11fa496ba48

The issue is that in git history there are also changes of images and datasets and I don't think it is necessary to have any of them in git history. The other con could be that in order to see git blame ppl would need to clone the shap_old in order to see it or to inspect it in github UI.
